httpie/downloads.py

```diff
@@ -180,6 +181,26 @@ def get_unique_filename(filename, exists=os.path.exists):
         attempt += 1
 
 
+def friendly_time(time):
+    '''
+    Takes in time in seconds and returns a more human friendly time.
+    '''
+    time_string = str(timedelta(seconds=time))
+    if 'day' in time_string or 'days' in time_string:  # Downloads for more than a day
+        time_split = time_string.split(', ')
+        hour_split = time_split[1].split(':')
+        hour_split[2] = float(hour_split[2])
+        return "{0} {1[0]}h {1[1]}m {1[2]:.2f}s".format(time_split[0], hour_split)
+    else:
+        time_split = time_string.split(':')
+        time_split[2] = float(time_split[2])
+        if int(time_split[0]):  # More than 1 hour but not a day
+            return "{0[0]}h {0[1]}m {0[2]:.2f}s".format(time_split)
+        elif int(time_split[1]):  # More than 1 minute but not an hour
+            return "{0[1]}m {0[2]:.2f}s".format(time_split)
+        return "{0[2]:.2f}s".format(time_split)  # Less than a min
+
+
```

Comment:
This seems rather complex for processing that information, especially with multiple branches that identify the same thing. Perhaps you can simplify it like this;
```suggestion
from datetime import timedelta

def friendly_time(total_seconds: float) -> str:
    '''
    Takes in time in seconds and returns a more human friendly time.
    '''
    time_string = str(timedelta(seconds=total_seconds))
    days, _, clock = time_string.rpartition(' days, ')
    hours, minutes, seconds = clock.split(':')
    
    output = ''
    for value, suffix in [
        (days, ' days,'),
        (hours, 'h'),
        (minutes, 'm'),
        (f'{float(seconds):0.2f}', 's'),
    ]:
        if value and float(value):
            output += value + suffix + ' '

    return output
```

Reply:
Even with this, I am not sure how I feel about string processing. Perhaps we can use this instead of manually parsing the output of timedelta;
```py
    days, remainder = divmod(total_seconds, 24 * 60 * 60)
    hours, remainder = divmod(remainder, 60 * 60)
    minutes, seconds = divmod(remainder, 60)
```

This way, we don't need to cast it back to `float()` again.